src/app/_components/client/Modal/components/ModalPortal/ModalPortal.tsx

```diff
@@ -0,0 +1,27 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { createPortal } from "react-dom";
+
+interface ModalPortalProps {
+  children: React.ReactNode;
+}
+
+const ModalPortal = ({ children }: ModalPortalProps) => {
+  const [isCSR, setIsCSR] = useState(false);
+
+  useEffect(() => {
+    setIsCSR(true);
+  }, []);
+
+  if (typeof window === "undefined") {
+    return <></>;
+  }
+  if (!isCSR) {
+    return <></>;
+  }
+
+  return createPortal(<>{children}</>, document.body);
```

Comment:
[ P4 ]

해당 코드를 아래처럼 변경할 수도 있을것같아서 올려봅니다

- body가 존재하지 않을 일은 잘없지만 state를 통해 검사할 필요가없을것같고, 아래 window에 대한 조건도 제거할 수 있을것같아요!
- 추가로 현재 코드 흐름으로는 isShow에 따라 매번 document가 불리는 상황은 동일할 것같지만, document를 사용하는 횟수가 최소화하는게 좋다고 하던데 body자체를 상태에 저장해놓고 저장해놓은 상태를 사용하여 doment사용을 줄일수도 잇을것같네요!

```suggestion
    const [BaseElement, setBaseElement] = useState<null | HTMLElement>(null);

  useEffect(() => {
    const findElement = document.body;
    if (!(findElement instanceof HTMLElement)) {
      return;
    }

    setBaseElement(findElement);
  }, []);

  if (!BaseElement) {
    return null;
  }

  return createPortal(<>{children}</>, BaseElement);
```

Reply:
해당 로직은 기존 코드에서 `state`, `effect` hook을 제거하여 처리하도록 수정했습니다 !

```jsx
if (typeof window === "undefined") {
    return <></>;
 }
```
해당 조건문으로 예외 처리만 진행해 주면 정상 동작하는 것 같습니다 😀